src/modules/launcher/Plugins/Microsoft.Plugin.Program/Programs/UWPApplication.cs

```diff
@@ -394,72 +378,277 @@ internal string LogoPathFromUri(string uri, string theme)
+            {
+                LogoPath = highContrastTargetIcon;
+                LogoType = LogoType.HighContrast;
+                return true;
+            }
+
+            var coloredScaleIcon = GetScaleIcons(path, theme);
+            if (!string.IsNullOrEmpty(coloredScaleIcon))
+            {
+                LogoPath = coloredScaleIcon;
+                LogoType = LogoType.Colored;
+                return true;
+            }
+
+            var coloredTargetIcon = GetTargetSizeIcon(path, theme);
+            if (!string.IsNullOrEmpty(coloredTargetIcon))
+            {
+                LogoPath = coloredTargetIcon;
+                LogoType = LogoType.Colored;
+                return true;
+            }
+
+            return false;
+        }
+
+        internal void LogoPathFromUri(string uri, Theme theme)
+        {
+            // all https://msdn.microsoft.com/windows/uwp/controls-and-patterns/tiles-and-notifications-app-assets
+            // windows 10 https://msdn.microsoft.com/en-us/library/windows/apps/dn934817.aspx
+            // windows 8.1 https://msdn.microsoft.com/en-us/library/windows/apps/hh965372.aspx#target_size
+            // windows 8 https://msdn.microsoft.com/en-us/library/windows/apps/br211475.aspx
+            string path;
+            bool isLogoUriSet;
+            if (uri.Contains("\\", StringComparison.Ordinal))
+            {
+                path = Path.Combine(Package.Location, uri);
+            }
+            else
+            {
+                // for C:\Windows\MiracastView etc
+                path = Path.Combine(Package.Location, "Assets", uri);
+            }
+
+            if (theme == Theme.HighContrastBlack || theme == Theme.HighContrastOne || theme == Theme.HighContrastTwo)
+            {
+                isLogoUriSet = SetHighContrastIcon(path, ContrastBlack);
+            }
+            else if (theme == Theme.HighContrastWhite)
+            {
+                isLogoUriSet = SetHighContrastIcon(path, ContrastWhite);
+            }
+            else if (theme == Theme.Light)
+            {
+                isLogoUriSet = SetColoredIcon(path, ContrastWhite);
+            }
+            else
+            {
+                isLogoUriSet = SetColoredIcon(path, ContrastBlack);
+            }
+
+            if (!isLogoUriSet)
+            {
+                LogoPath = string.Empty;
+                LogoType = LogoType.Error;
+                ProgramLogger.LogException(
+                            $"|UWP|LogoPathFromUri|{Package.Location}" +
+                            $"|{UserModelId} can't find logo uri for {uri} in package location: {Package.Location}", new FileNotFoundException());
+            }
+        }
+
         public ImageSource Logo()
         {
-            var logo = ImageFromPath(LogoPath);
-            return logo;
+            if (LogoType == LogoType.Colored)
+            {
+                var logo = ImageFromPath(LogoPath);
```

Comment:
I may have been mistaken but did we want to add a background color for all the icons? I thought it was only for the icons like Spotify which were appearing white on white in light theme. @crutkas, @niels9001

Reply:
I believe the idea was to match the way icons are displayed in start menu instead of displaying their high contrast variants.